plugins/spectate/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+
+project(spectate)
+
+SET(SOURCES
+        spectate.cpp
+        pause.cpp)
+
```

Comment:
You should probably mark your header file here too so changes to the header file will trigger recompliation. Here's how labormanager does it: 
```
set(PROJECT_HDRS
    labormanager.h
    joblabormapper.h
)
set_source_files_properties(${PROJECT_HDRS} PROPERTIES HEADER_FILE_ONLY TRUE)

# mash them together (headers are marked as headers and nothing will try to compile them)
list(APPEND PROJECT_SRCS ${PROJECT_HDRS})
```

Reply:
The only thing this really does is cause the headers to be listed in the same "project", like in a Visual Studio sidebar.

CMake detects header dependencies automatically, e.g. with `gcc -M`. If you touch `joblabormapper.h` and rebuild, you'll see the same things being rebuilt with and without `joblabormapper.h` commented out in the labormanager CMakeLists, for instance.